guide/src/guide/new-cell.md

```diff
@@ -1 +1,221 @@
+
+  ```rust
+  #[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+  #[derive(Debug, Default, Clone)]
+  pub struct LosePacketCellConfig {
+      pub loss_interval: usize,
+  }
+
+  impl LosePacketCellConfig {
+      pub fn new<T: Into<usize>>(loss_interval: T) -> Self {
+          Self {
+              loss_interval: loss_interval.into(),
+          }
+      }
+  }
+  ```
+
+- `CellControlInterface`
+
+  `CellControlInterface` provides a way to control inner parameters for your cell while it is running. `ControlInterface` trait must be implemented for this struct.
+
+  ```rust
+  pub struct LosePacketCellControlInterface {
+      /// Stored as nanoseconds
+      loss_interval: Arc<AtomicRawCell<usize>>,
+  }
+
+  impl ControlInterface for LosePacketCellControlInterface {
+      type Config = LosePacketCellConfig;
+  
+      fn set_config(&self, config: Self::Config) -> Result<(), Error> {
+          self.loss_interval.store(Box::new(config.loss_interval));
+          Ok(())
+      }
+  }
+  ```
+
+- `Cell`
+
+  `Cell` combinates all these four structs to make a whole cell.
+
+  ```rust
+  pub struct LosePacketCell<P: Packet> {
+      ingress: Arc<LosePacketCellIngress<P>>,
+      egress: LosePacketCellEgress<P>,
+      control_interface: Arc<LosePacketCellControlInterface>,
+  }
+
+  impl<P> Cell<P> for LossCell<P>
+  where
+      P: Packet + Send + Sync + 'static,
+  {
+      type IngressType = LosePacketCellIngress<P>;
+      type EgressType = LosePacketCellEgress<P>;
+      type ControlInterfaceType = LosePacketCellControlInterface;
+
+      fn sender(&self) -> Arc<Self::IngressType> {
+          self.ingress.clone()
+      }
+
+      fn receiver(&mut self) -> &mut Self::EgressType {
+          &mut self.egress
+      }
+
+      fn into_receiver(self) -> Self::EgressType {
+          self.egress
+      }
+
+      fn control_interface(&self) -> Arc<Self::ControlInterfaceType> {
+          Arc::clone(&self.control_interface)
+      }
+  }
+
+  impl<P> LosePacketCell<P>
+  where
+      P: Packet,
+  {
+      pub fn new<L: Into<usize>>(loss_interval: L) -> Result<LosePacketCell<P>, Error> {
+          let loss_interval = loss_interval.into();
+          let (rx, tx) = mpsc::unbounded_channel();
+          let loss_interval = Arc::new(AtomicRawCell::new(Box::new(loss_interval)));
+          Ok(LosePacketCell {
+              ingress: Arc::new(LosePacketCellIngress { ingress: rx }),
+              egress: LosePacketCellEgress {
+                  egress: tx,
+                  loss_interval: Arc::clone(&loss_interval),
+                  inner_loss_interval: Box::default(),
+                  state: AtomicI32::new(0),
+                  counter: 0,
+              },
+              control_interface: Arc::new(LosePacketCellControlInterface { loss_interval }),
+          })
+      }
+  }
+  ```
+
+## Add Configuration to Your New Cell
+
+- TODO: How to add configuration support to a new cell(In tests and in .toml files)
+
```

Comment:
We should state the premise of this section: if you want to integrate the cell into the rattan official repository to the corresponding CLI tool.